main.py

```diff
@@ -160,6 +160,39 @@ def string2timestamp(s):
 def year2timestamp(y):
     return time.mktime(datetime.date(y, 1, 1).timetuple())
 
+@app.route('/new-features', methods=['POST'])
+def updatebyemail():
+    emailaddress = str(request.json['address'])
+    fname = str(request.json['fname'])
+    lname = str(request.json['lname'])
+
+    if len(fname)>40:
+        return  jsonify(respo='First name to long')
+    if len(lname)>40:
+        return  jsonify(respo='Last name to long')
+    if len(emailaddress)>40:
+        return jsonify(respo='Email too long', emailaddress = emailaddress)
+    user_exists = db_session.query(User).filter(and_(User.first_name == fname, User.last_name == lname,
+                                                 User.email == emailaddress))
+    if user_exists.count()==0:
+        user = User(email = emailaddress, first_name = fname, last_name = lname, new_features_subscription=True)
+        db_session.add(user)
+        db_session.commit()
+        return jsonify(respo='Subscription saved', )
+    else:
+        subscription_exists = User.query.filter(and_(User.first_name == fname, User.last_name == lname,
```

Comment:
I guess you are right. The email is the best unique identifier here. I will change it accordingly.